src/app/+dashboard/components/greeting/greeting.component.ts

```diff
@@ -0,0 +1,25 @@
+import {Component} from '@angular/core';
+
+@Component({
+  selector: 'app-greeting',
+  templateUrl: './greeting.component.html',
+  styleUrls: ['./greeting.component.scss']
+})
+export class GreetingComponent {  
+
+  public greeting = this.dayTime();
+
+  dayTime(): string {
+    const hour = new Date().getHours();
+
+    if (hour >= 5 && hour <= 11) {
+      return 'Good Morning, ';
+    } else if (hour >= 12 && hour <= 17) {
+      return 'Good Day, ';
+    } else if (hour >= 18 && hour <= 23) {
+      return 'Good Evening, ';
+    } else {
+      return 'Good Night, ';
+    }
+  }
+}
```

Comment:
Here are some suggestions to how improve your code here. Please take a look.

```suggestion
export class GreetingComponent implements OnInit {  
  greeting: string;
  
  ngOnInit(): void {
     this.greeting = this.getDayTime();
  }

  private getDayTime(): string {
    const hour = new Date().getHours();

    if (hour >= 5 && hour <= 11) {
      return 'Good Morning, ';
    } else if (hour >= 12 && hour <= 17) {
      return 'Good Day, ';
    } else if (hour >= 18 && hour <= 23) {
      return 'Good Evening, ';
    } else {
      return 'Good Night, ';
    }
  }
}
```

Reply:
fixed